### PR TITLE
lsof: update to 4.91 and remove download strategy.

### PR DESCRIPTION
**Pull request by reitermarkus**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

### CODE REVIEW

**Comment by ilovezfs** [comment]
What's the purpose of this change? See extended prior conversation in https://github.com/Homebrew/homebrew-dupes/pull/616.

**Comment by reitermarkus** [comment]
I really disagree that subclassing a download strategy is cleaner than `inreplace` in this case, because firstly, `patch` not working in this case has nothing to do with the download strategy, and secondly, download strategies should be treated as a private API. After https://github.com/Homebrew/homebrew-core/pull/30659 and this PR, there will be no formula using custom download strategies, which is a good thing.

File: Formula/lsof.rb [hunk 1]
```diff
@@ -21,12 +12,13 @@ class Lsof < Formula
...
   end
 
-  patch do
-    url "https://raw.githubusercontent.com/Homebrew/formula-patches/c3acbb8/lsof/lsof-489-darwin-compile-fix.patch"
-    sha256 "997d8c147070987350fc12078ce83cd6e9e159f757944879d7e4da374c030755"
-  end
-
   def install
+    system "/usr/bin/tar", "xf", "#{name}_#{version}_src.tar"
+    cd "#{name}_#{version}_src"
```
Comment (ilovezfs) [inline Formula/lsof.rb]: This should use `cd "#{name}_#{version}_src" do`

File: Formula/lsof.rb [hunk 2]
```diff
@@ -21,12 +12,13 @@ class Lsof < Formula
     sha256 "8a63b65f74b992e4a70ac3e18a7e4b810a73e183b13086b40f30286256e646e0" => :yosemite
   end
 
-  patch do
-    url "https://raw.githubusercontent.com/Homebrew/formula-patches/c3acbb8/lsof/lsof-489-darwin-compile-fix.patch"
-    sha256 "997d8c147070987350fc12078ce83cd6e9e159f757944879d7e4da374c030755"
-  end
-
   def install
+    system "/usr/bin/tar", "xf", "#{name}_#{version}_src.tar"
```
Comment (ilovezfs) [inline Formula/lsof.rb]:
```
system "tar", "xf", "#{name}_#{version}_src.tar"
```
instead of `/usr/bin/tar`

**Review by ilovezfs — DISMISSED** [review]
Due to the removal of the download strategy, the install method now needs an explicit `prefix.install_metafiles` otherwise the README file won't get installed.

File: Formula/lsof.rb (hunk 1 above)
Comment (ilovezfs) [reply Formula/lsof.rb]: Actually let's hard code `lsof` instead of using `#{name}`

File: Formula/lsof.rb [hunk 3]
```diff
@@ -21,26 +12,28 @@ class Lsof < Formula
...
   end
 
-  patch do
-    url "https://raw.githubusercontent.com/Homebrew/formula-patches/c3acbb8/lsof/lsof-489-darwin-compile-fix.patch"
-    sha256 "997d8c147070987350fc12078ce83cd6e9e159f757944879d7e4da374c030755"
-  end
-
   def install
-    ENV["LSOF_INCLUDE"] = "#{MacOS.sdk_path}/usr/include"
+    system "tar", "xf", "#{name}_#{version}_src.tar"
```
Comment (ilovezfs) [inline Formula/lsof.rb]: just use lsof instead of `name`

File: Formula/lsof.rb [hunk 4]
```diff
@@ -21,26 +12,28 @@ class Lsof < Formula
...
+    cd "#{name}_#{version}_src" do
```
Comment (ilovezfs) [inline Formula/lsof.rb]: ditto

**Comment by ilovezfs** [comment]
shipped! Thanks @reitermarkus

**Pull request closed without merging**